engine/consensus/approvals/assignment_collector_tree.go

```diff
@@ -118,13 +120,20 @@ func (t *AssignmentCollectorTree) FinalizeForkAtLevel(finalized *flow.Header, se
 		}
 
 		if len(finalizedFork) > 0 {
-			if !finalizedFork[0].processable {
+			if finalizedFork[0].collector.ProcessingStatus() != VerifyingApprovals {
```

Comment:
I am confused, can we have multiple _finalized_ forks? This code is basically for a case where finalized fork has to be verifiable otherwise we have a bug somewhere.

Reply:
We need to distinguish between the tree of blocks and the tree of results (aka execution tree)
* tree of blocks: we can only have a single finalized fork
* execution tree: for the finalized blocks, there can still be conflicting branches of results in the execution tree. 
   * Sealing orphans forks in the execution tree.
   * But since the finalized blocks are still unsealed, there can be multiple competing results